src/apps/checklist-button/checklist-material-button.slice.jsx

```diff
@@ -40,31 +53,44 @@ export const checklistMaterialButtonSlice = createSlice({
   name: "checklistMaterial",
   initialState: { status: {}, onList: {} },
   reducers: {
+    setInitialStatus(state, action) {
+      state.status[action.payload.materialId] = "ready";
+      state.onList[action.payload.materialId] = action.payload.onList;
+    },
     addToListPending(state, action) {
       state.status[action.payload.materialId] = "pending";
-      state.onList[action.payload.materialId] = "on";
+      state.onList[action.payload.materialId] = "off";
     },
     addToListAborted(state, action) {
       state.status[action.payload.materialId] = "failed";
       state.onList[action.payload.materialId] = "off";
     },
     removeFromListPending(state, action) {
       state.status[action.payload.materialId] = "pending";
-      state.onList[action.payload.materialId] = "off";
+      state.onList[action.payload.materialId] = "on";
     },
     removeFromListAborted(state, action) {
       state.status[action.payload.materialId] = "failed";
       state.onList[action.payload.materialId] = "on";
     }
   },
   extraReducers: {
+    [checkOnListAction.pending]: (state, action) => {
+      state.onList[action.meta.arg.materialId] = "unknown";
+    },
+    [checkOnListAction.fulfilled]: (state, action) => {
+      state.onList[action.meta.arg.materialId] = action.payload ? "on" : "off";
+    },
+    [checkOnListAction.rejected]: (state, action) => {
+      state.onList[action.meta.arg.materialId] = "off";
+    },
     [addToListAction.pending]: (state, action) => {
       state.status[action.meta.arg.materialId] = "processing";
-      state.onList[action.meta.arg.materialId] = "on";
+      state.onList[action.meta.arg.materialId] = "off";
     },
     [addToListAction.fulfilled]: (state, action) => {
       state.status[action.meta.arg.materialId] = "finished";
-      state.onList[action.meta.arg.materialId] = "on";
+      state.onList[action.meta.arg.materialId] = "off";
```

Comment:
Some of these `onList` states mess with my head. 

With optimistic UI I would have expected that the material should be handled as `on` the list when an `addToListAction` is pending.

In general I would expect that the material should be `on` the list when an `addToListAction` is fulfilled.